MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/labs/RecyclerViewOnMapActivity.java

```diff
@@ -0,0 +1,275 @@
+package com.mapbox.mapboxandroiddemo.labs;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.CardView;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.annotations.MarkerOptions;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Use a recyclerview with a Mapbox map to easily explore content all on one screen
+ */
+public class RecyclerViewOnMapActivity extends AppCompatActivity {
+
+  private MapView mapView;
+  public MapboxMap mapboxMap;
+  private RecyclerView recyclerView;
+  private LocationRecyclerViewAdapter locationAdapter;
+  private ArrayList<SingleRecyclerViewLocation> locationList;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_lab_recycler_view_on_map);
+
+    recyclerView = (RecyclerView) findViewById(R.id.rv_on_top_of_map);
+
+    Toast.makeText(this, R.string.toast_instruction, Toast.LENGTH_SHORT).show();
+
+    // Initialize the map view
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+        RecyclerViewOnMapActivity.this.mapboxMap = mapboxMap;
+
+        setUpLists();
+
+        // Set up the recyclerView
+        locationAdapter = new LocationRecyclerViewAdapter(locationList, mapboxMap);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext(),
+            LinearLayoutManager.HORIZONTAL, true));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(locationAdapter);
```

Comment:
This is a great opportunity to use a [the SnapHelper](https://developer.android.com/reference/android/support/v7/widget/SnapHelper.html) class. Would look something like this:
```
SnapHelper snapHelper = new LinearSnapHelper();
snapHelper.attachToRecyclerView(recyclerView);
```